### PR TITLE
WFLY-11913 JGroups client sockets are not registered with the socket binding manager

### PR DESCRIPTION
**Pull request by pferraro**
Builds upon https://github.com/wildfly/wildfly/pull/12188

Adds socket bindings for client sockets for TCP-based transports and FD_SOCK, which bind to a random ephemeral port, by default.

Requires wildfly-core upgrade containing https://github.com/wildfly/wildfly-core/pull/3726

### CODE REVIEW

**Comment by bstansberry** [comment]
retest this please

**Comment by pferraro** [comment]
This PR seems to be causing a lot of intermittent failures in a number of clustering tests.  Adding hold label until a cause is identified.

File: clustering/jgroups/extension/src/main/resources/org/jboss/as/clustering/jgroups/subsystem/LocalDescriptions.properties [hunk 1]
```diff
@@ -94,7 +94,8 @@ jgroups.protocol.data-source=Data source reference for JDBC_PING protocol to be
 jgroups.protocol.type=The implementation class for a protocol, which determines protocol functionality.
 jgroups.protocol.type.deprecated=Deprecated. Will be removed in a future version.
 jgroups.protocol.module=The module with which to resolve the protocol type.
-jgroups.protocol.socket-binding=The socket binding specification for this protocol layer, used to specify IP interfaces and ports for communication.
+jgroups.protocol.socket-binding=Defines the bind/address port used of the server socket used to receive messages from other cluster members.
```
Comment (rhusar) [inline clustering/jgroups/extension/src/main/resources/org/jboss/as/clustering/jgroups/subsystem/LocalDescriptions.properties]: Should be "Defines the bind address/port"...

File: clustering/jgroups/extension/src/main/resources/schema/jboss-as-jgroups_7_0.xsd [hunk 2]
```diff
@@ -0,0 +1,546 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+  ~ JBoss, Home of Professional Open Source.
+  ~ Copyright 2016, Red Hat, Inc., and individual contributors
```
Comment (rhusar) [inline clustering/jgroups/extension/src/main/resources/schema/jboss-as-jgroups_7_0.xsd]: 2019

File: clustering/jgroups/extension/src/test/resources/org/jboss/as/clustering/jgroups/subsystem/subsystem-jgroups-7_0.xml [hunk 3]
```diff
@@ -0,0 +1,84 @@
+<!--
+  ~ JBoss, Home of Professional Open Source.
+  ~ Copyright 2016, Red Hat, Inc., and individual contributors
```
Comment (rhusar) [inline clustering/jgroups/extension/src/test/resources/org/jboss/as/clustering/jgroups/subsystem/subsystem-jgroups-7_0.xml]: 2019

**Review by rhusar — APPROVED** [review]
LGTM, just a few cosmetic changes.

**Review by bstansberry — APPROVED** [review]

**Pull request merged**